src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -183,13 +183,15 @@ private bool IsUnusedParameterCandidate(IParameterSymbol parameter)
                 // we don't have to bail out on reporting unused parameters for all public methods.
 
                 if (parameter.IsImplicitlyDeclared ||
+                    parameter.Name.Length == 0 ||
                     parameter.Name == DiscardVariableName ||
                     !(parameter.ContainingSymbol is IMethodSymbol method) ||
                     method.IsImplicitlyDeclared ||
                     method.IsExtern ||
                     method.IsAbstract ||
                     method.IsVirtual ||
                     method.IsOverride ||
+                    method.PartialImplementationPart != null ||
```

Comment:
i hate this entire check because it's just a blob of conditions, with no explanation on any of it as to why they're relevant.